src/Services/CoinSelectionService.cs

```diff
@@ -113,13 +115,58 @@ private IBitcoinRequestRepository GetRepository(BitcoinRequestType requestType)
     public async Task LockUTXOs(List<UTXO> selectedUTXOs, IBitcoinRequest bitcoinRequest, BitcoinRequestType requestType)
     {
         // We "lock" the PSBT to the channel operation request by adding to its UTXOs collection for later checking
-        var utxos = selectedUTXOs.Select(x => _mapper.Map<UTXO, FMUTXO>(x)).ToList();
+        var selectedFMUTXOs = selectedUTXOs.Select(x => _mapper.Map<UTXO, FMUTXO>(x)).ToList();
+        var utxos = await _fmutxoRepository.GetByWalletId(bitcoinRequest.Wallet.Id);
 
-        var addUTXOsOperation = await GetRepository(requestType).AddUTXOs(bitcoinRequest, utxos);
+        var utxosToSave = new List<FMUTXO>();
+        var utxosToAddToTransaction = new List<FMUTXO>();
+        // We check the selected UTXOs against the ones we have in the database to see if they are already there
```

Comment:
nit:
```suggestion
        // Since the UTXOs to lock (selectedUTXOs) come directly from nbxplorer, we check if we already registered them in our database with the RegisterUTXOsJob
```